### ISSUE TITLE
OpenType feature support for better cartography

### ISSUE THREAD

**Issue by Chaz6** [qgis/QGIS#32733]
**Feature description.**
Please consider supporting OpenType features:-

- Stylistic alternates
- Tabular figures
- Ligatures (regular and discretionary)

There are others, but I think these three would have the most impact on cartographic presentation. As with other options, these could be data-driven (enabled or disabled)

**Additional context**
For more information on opentype features, the following urls have more information:-
https://typofonderie.com/font-support/opentype-features/
https://en.wikipedia.org/wiki/List_of_typographic_features#OpenType_typographic_features
https://ilovetypography.com/OpenType/opentype-features.html

**Comment by andreasneumann** [comment]
Hi @Chaz6 

This all depends on support in qt5. If qt5 supports it, then it would be realistic to add such features, if not, it would have to be added in qt first.

If you can help to investigate if the desired features are supported in qt, it would help.

We recently started collaborating with company KDAB to do fixes and improvements in qt5 upstream. So we are slowly trying to improve collaboration with qt - because - a lot in QGIS depends on availability in qt.

**Comment by jgrocha** [comment]
As far as I know, QGIS supports OpenType fonts (provided by the underneath software layers).

For testing OpenType fonts, I used the [Libertinus](https://github.com/alif-type/libertinus) font. In the following screen shot, you can see the typical **fi** ligature in the word **filipe**.

![QGIS with OpenType ligatures](https://user-images.githubusercontent.com/163681/68527547-e4ee3380-02df-11ea-9b9a-adf0cc649d51.png)

I've downloaded another [OpenTyle font](https://www.fontspace.com/category/ligatures,opentype), installed it and created a label with that font. It works.

![QGIS with OpenType](https://user-images.githubusercontent.com/163681/68527557-00593e80-02e0-11ea-8b9a-59e1b7e7d65b.png)

@Chaz6 Do you want to turn on/turn off OpenType features for each font? That's the kind of features you are looking for, like we can do in [LibreOffice](http://www.scholarsfonts.net/Using_OT_in_LibreOffice_1-2.pdf)?

**Comment by Chaz6** [comment]
@jgrocha yes I think it would be useful to be able to have a bit more control, such as choosing stylistic sets. I could have a go at writing a patch.

**Comment by lliehu** [comment]
It looks like Qt 6.6 (to be released later this year) has added support for OpenType features to `QFont`, acording to these sources:
* [Qt bug report](https://bugreports.qt.io/browse/QTBUG-89958)
* [QFont API documentation (snapshot)](https://doc-snapshots.qt.io/qt6-6.6/qfont.html#setFeature-1)
* [What's New in Qt 6.6 (snapshot)](https://doc-snapshots.qt.io/qt6-6.6/whatsnew66.html)

Perhaps this would enable support in QGIS, after QGIS can otherwise depend on Qt 6.
